kernel/build-kernel.sh

```diff
@@ -128,19 +130,77 @@ get_kernel() {
 	mv "linux-${version}" "${kernel_path}"
 }
 
+# Make a kernel config file from generic and arch specific
+# fragments
+# - arg1 - path to arch specific fragments
+# - arg2 - path to kernel sources
+#
+get_kernel_frag_path() {
+	local arch_path="$1"
+	local common_path="${arch_path}/../common"
+	local kernel_path="$2"
+	local cmdpath="${kernel_path}/scripts/kconfig/merge_config.sh"
+	local config_path="${arch_path}/.config"
+
+	local arch_configs="$(ls ${arch_path}/*.conf)"
+	local common_configs="$(ls ${common_path}/*.conf)"
+	local all_configs="${common_configs} ${arch_configs}"
+
+	local results=$(export KCONFIG_CONFIG=${config_path}; export ARCH=${arch_target}; cd ${kernel_path}; ${cmdpath} -r -n ${all_configs})
+
+	# Did we request any entries that did not make it?
+	local missing=$(grep -v -q "not in final" <<< ${results}; echo $?)
```

Comment:
With my paranoia hat on, I wonder if it would make sense to set `LC_ALL=C` before running `cmdpath` *just in case* it ever gets i18n'd. If it did, this may fail mysteriously. Atleast it would on my system ;)

Reply:
too much coffee? :-) a bit paranoid indeed...  I'm going to say we can cross that bridge if we ever get to it (we probably have many many such cases through our codebase) - but, feel free to post-PR an update if you want :-)

Reply:
... but, but, but... what about parity errors, aliens, lightning strikes, cosmic rays? :)

Yeah, we should be able to deal with this at a higher level.